--- a/Source/cmake.cxx
+++ b/Source/cmake.cxx
@@ -130,8 +130,10 @@
 // NOTE: the __linux__ macro is predefined on Android host too, but
 // main CMakeLists.txt filters out this generator by host name.
+#if !defined(CMAKE_BOOTSTRAP)
 #if (defined(__linux__) && !defined(__ANDROID__)) || defined(_WIN32)
 #  include "cmGlobalGhsMultiGenerator.h"
 #endif
+#endif

 #if defined(__APPLE_DISABLED__)
 #  if !defined(CMAKE_BOOTSTRAP)
